biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java

```diff
@@ -277,19 +284,74 @@ void doAnnotatedAnnotation(final Annotation annotation, TypeRef name) throws Exc
 			}
 			if (scanThisType) {
 				c.parseClassFileWithCollector(new ClassDataCollector() {
+					private MethodDef			lastMethodSeen;
+
+					private Attrs	attributesAndDirectives	= new Attrs();
+					
 					@Override
 					public void annotation(Annotation a) throws Exception {
+						if (ATTRIBUTE.equals(a.getName().getFQN()) || 
+							DIRECTIVE.equals(a.getName().getFQN())) {
+							handleAttributeOrDirective(a);
+						}
+
 						if (interesting.contains(a.getName().getFQN())) {
 							// Bnd annotations support merging of child properties,
 							// but this is not in the specification as far as I can tell
 							if(isBndAnnotation(a)) {
 								a.merge(annotation);
 								a.addDefaults(c);
+							} else if (isRequirementOrCapability(a)) {
+								mergeAttributesAndDirectives(a);
 							}
 							AnnotationHeaders.this.annotation(a);
 						}
 					}
 
+					private void mergeAttributesAndDirectives(Annotation a) {
+						if(STD_CAPABILITIES.equals(a.getName().getFQN()) || STD_REQUIREMENTS.equals(a.getName().getFQN())) {
+							Object[] annotations = a.get("value");
+							for (int i = 0; i < annotations.length; i++) {
+								mergeAttributesAndDirectives((Annotation) annotations[i]);
+							}
+						} else {
+							attributesAndDirectives.entrySet()
```

Comment:
Dead code? (Sorry, trying to fix the comment to reflect the 3 lines I am referencing)

Reply:
Yes, it does seem dead.